Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.LoopExpression.cs

```diff
@@ -49,8 +49,11 @@ protected internal override Expression Accept (ExpressionVisitor visitor)
       return default(Expression);
     }
 
+    [Pure]
     public LoopExpression Update (LabelTarget breakLabel, LabelTarget continueLabel, Expression body)
     {
+      Contract.Requires(body != null);
+      Contract.Ensures(Contract.Result<LoopExpression>() != null);
```

Comment:
From this line of code: http://referencesource.microsoft.com/#System.Core/Microsoft/Scripting/Ast/LoopExpression.cs,137

Additional requies could be added: `Contract.Requies(continueLabel == null || continueLabel.Type != typeof(void), "Label type must be void");`